src/generators/shared.ts

```diff
@@ -0,0 +1,248 @@
+import { ISanitizer } from '@jupyterlab/apputils';
+import { Cell } from '@jupyterlab/cells';
+import { IHeading } from '../toc';
+
+const VDOM_MIME_TYPE = 'application/vdom.v1+json';
+
+const HTML_MIME_TYPE = 'text/html';
+
+export interface INotebookHeading extends IHeading {
+  numbering?: string | null;
+  type: string;
```

Comment:
This `type` is confusing to me. What kinds of things can it be? If it is only a small number of things, it should probably be a union type. One of its settings above is `'header''. Aren't they all headers?`